api/src/main/java/jakarta/data/page/Page.java

```diff
@@ -17,39 +17,193 @@
+     *
+     * @return a request for the next page.
+     * @throws NoSuchElementException if it is known that there is no next page.
+     *         To avoid this exception, check for a {@code true} result of
+     *         {@link #hasNext()} before invoking this method.
+     */
+    PageRequest<T> nextPageRequest();
+
+    /**
+     * <p>Returns a request for the {@linkplain PageRequest#next() next} page if
+     * {@link #hasNext()} indicates there might be a next page.</p>
+     *
+     * <p>This method is useful when the {@linkplain Query query language} is used
+     * to return a result of different type to the entity class that was queried,
+     * allowing the subsequent {@code PageRequest} to be cast back to the type of
+     * entity class that was queried.</p>
+     *
+     * @param <E>         entity class of the attributes that are used as sort criteria.
+     * @param entityClass entity class of the attributes that are used as sort criteria.
+     * @return a request for the next page.
+     * @throws NoSuchElementException if it is known that there is no next page.
+     *         To avoid this exception, check for a {@code true} result of
+     *         {@link #hasNext()} before invoking this method.
+     */
+    <E> PageRequest<E> nextPageRequest(Class<E> entityClass);
+
+    /**
+     * <p>Returns a request for the {@link PageRequest#previous() previous} page,
+     * if {@link #hasPrevious()} indicates there might be a previous page.</p>
+     *
+     * @return a request for the previous page.
+     * @throws NoSuchElementException if it is known that there is no previous page.
+     *         To avoid this exception, check for a {@code true} result of
+     *         {@link #hasPrevious()} before invoking this method.
+     */
+    PageRequest<T> previousPageRequest();
+
+    /**
+     * <p>Returns a request for the {@linkplain PageRequest#previous() previous}
+     * page, if {@link #hasPrevious()} indicates there might be a previous page.</p>
+     *
+     * <p>This method is useful when the {@linkplain Query query language} is used
+     * to return a result of different type to the entity class that was queried,
+     * allowing the subsequent {@code PageRequest} to be cast back to the type of
+     * entity class that was queried.</p>
+     *
+     * @param <E>         entity class of the attributes that are used as sort criteria.
+     * @param entityClass entity class of the attributes that are used as sort criteria.
+     * @return a request for the previous page.
+     * @throws NoSuchElementException if it is known that there is no previous page.
+     *         To avoid this exception, check for a {@code true} result of
+     *         {@link #hasPrevious()} before invoking this method.
+     */
+    <E> PageRequest<E> previousPageRequest(Class<E> entityClass);
+
+    /**
+     * Returns {@code true} if the {@link #pageRequest()} specified that the
```

Comment:
Pagination with total is a common feature in many NoSQL databases, but it's **not universal** across all NoSQL databases. Whether pagination with total is supported depends on the specific database management system and its features.

I would go with Unsupportedoperationexception when not supported.